packages/kotti-ui/source/kotti-field-text-area/hooks.ts

```diff
@@ -0,0 +1,34 @@
+import { Ref, watch } from '@vue/composition-api'
+
+import { KottiFieldTextArea } from './types'
+
+const resizeTextarea = (
+	textarea: HTMLTextAreaElement | null,
+	autoSize: boolean,
+	maxHeight: number | null,
+) => {
+	if (
+		autoSize &&
+		textarea &&
+		((maxHeight && textarea?.scrollHeight < maxHeight) || !maxHeight)
+	) {
+		textarea?.setAttribute('style', 'height: auto')
+		textarea?.setAttribute('style', `height: ${textarea.scrollHeight}px`)
+	}
+}
+
+export const useResizeTextarea = (
+	textareaRef: Ref<HTMLTextAreaElement | null>,
+	value: Ref<KottiFieldTextArea.PropsInternal['value']>,
+	autoSize: Ref<boolean>,
+	maxHeight: Ref<number | null>,
+) => {
+	watch(
+		[textareaRef, value],
+		() => resizeTextarea(textareaRef.value, autoSize.value, maxHeight.value),
+		{
+			immediate: true,
+			flush: 'post',
+		},
+	)
+}
```

Comment:
maxHeight should communicate the expected unit, pixels in this case, by either updating the name of the variable or with `JSDoc` comments

```suggestion
const resizeTextarea = (textarea: HTMLTextAreaElement) => {
	textarea.setAttribute('style', 'height: auto')
	textarea.setAttribute('style', `height: ${textarea.scrollHeight}px`)
}

export const useResizeTextarea = (
	textareaRef: Ref<HTMLTextAreaElement | null>,
	value: Ref<KottiFieldTextArea.PropsInternal['value']>,
	autoSize: Ref<boolean>,
	maxHeightInPixels: Ref<number | null>,
) => {
	watch(
		[textareaRef, value],
		() => {
			if (!textareaRef.value) return

			const shouldResize =
				autoSize.value &&
				(!maxHeightInPixels.value ||
					textareaRef.value.scrollHeight <= maxHeightInPixels.value)

			if (shouldResize) resizeTextarea(textareaRef.value)
		},
		{
			immediate: true,
			flush: 'post',
		},
	)
}
```